warehouse/sessions.py

```diff
@@ -140,6 +145,22 @@ def invalidate(self):
     def should_save(self):
         return self._changed
 
+    def record_auth_timestamp(self):
+        self[self._reauth_timestamp_key] = str(datetime.datetime.now().timestamp())
+        self.changed()
+
+    def needs_reauthentication(self):
+        reauth_timestamp = self.get(self._reauth_timestamp_key)
+        if reauth_timestamp is None:
+            return True
+
+        try:
+            auth_time = float(reauth_timestamp)
+            current_time = datetime.datetime.now().timestamp()
+            return current_time - auth_time >= self.time_to_reauth
+        except (ValueError, TypeError):
+            return True
```

Comment:
When would this happen? Do we actually need to handle these exceptions or can we just guard against them?

Reply:
`TypeError` doesn't need to be there since I already handled the case where the timestamp is `None`. A `ValueError` could be raised if the timestamp can't be cast to a float for some reason, though I can't see how that field would ever get corrupted. So yeah, it should be fine to just guard against the `None` case, which I already do.